### PR TITLE
[WIP] Improve widget api

### PR DESCRIPTION
**Pull request by pahans**
## Purpose
Improve widget API

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx [hunk 1]
```diff
@@ -17,17 +17,17 @@
...
+    FormattedMessage,
 } from 'react-intl';
-import Axios from 'axios';
 import Moment from 'moment';
 import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
 import CircularProgress from '@material-ui/core/CircularProgress';
 import Paper from '@material-ui/core/Paper';
 import Typography from '@material-ui/core/Typography';
-import Widget from '@wso2-dashboards/widget';
+import withWrappedWidget from '@analytics-apim/common-lib/src/WrappedWidget';
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:30]: Does this import impacts new widget generator script?

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx [hunk 2]
```diff
@@ -51,26 +51,15 @@ const lightTheme = createMuiTheme({
...
+let refreshIntervalId = null;
+const refreshInterval = 60000; // 1min
 
 /**
  * Create React Component for APIM Api Created
  * @class APIMApiUsageSummaryWidget
  * @extends {Widget}
  */
-class APIMApiUsageSummaryWidget extends Widget {
+class APIMApiUsageSummaryWidget extends Component {
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:62]: If we get rid of base widget inheritance, how does the pub-sub model, global state sharing etc works?

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx [hunk 3]
```diff
@@ -291,35 +213,53 @@ class APIMApiUsageSummaryWidget extends Widget {
...
-                </MuiThemeProvider>
-            </IntlProvider>
+            <MuiThemeProvider theme={themeName === 'dark' ? darkTheme : lightTheme}>
+                {
+                    faultyProviderConf ? (
+                        <div style={paperWrapper}>
+                            <Paper elevation={1} style={paper}>
+                                <Typography variant='h5' component='h3'>
+                                    <FormattedMessage
+                                        id='config.error.heading'
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:223]: Have we tested and verified localization is working fine after this refactoring?

File: components/org.wso2.analytics.apim.widgets/CommonLib/src/WrappedWidget/index.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/CommonLib/src/WrappedWidget/index.jsx:2]: Wrong license year

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/stories/widget.stories.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable react/jsx-filename-extension */
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/stories/widget.stories.js:1]: Missing license

File: components/org.wso2.analytics.apim.widgets/CommonLib/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (TanyaM) [inline components/org.wso2.analytics.apim.widgets/CommonLib/.eslintrc.js:2]: wrong license year

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 1 above)
Comment (pahans) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:30]: We need to migrate that as well. But the existing code will work for now.

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 2 above)
Comment (pahans) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:62]: moved widget inheritance to common util, This should be used for every widget. https://github.com/wso2/analytics-apim/pull/1106/files#diff-9727f5b0fe8883c05f0e8a0469df9f6d

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 3 above)
Comment (pahans) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:223]: checked the wrapped widget util. https://github.com/wso2/analytics-apim/pull/1106/files#diff-9727f5b0fe8883c05f0e8a0469df9f6d

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 2 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:62]: But this doesn't have pub-sub and global state implementations.

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 3 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:223]: I saw the implementation. Just wanted to verify whether we tested.

File: components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx (hunk 2 above)
Comment (pahans) [reply components/org.wso2.analytics.apim.widgets/APIMApiUsageSummary/src/APIMApiUsageSummaryWidget.jsx:62]: line 154

**Pull request closed without merging**